src/checksum.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::vec::Vec;
+use libc::size_t;
+use Checksum;
+use ffi;
+use translate::*;
+
+impl Checksum {
+    pub fn get_digest(self) -> Vec<u8> {
+        unsafe {
+            let mut digest_len: size_t = 4096;
+            let mut vec = Vec::with_capacity(digest_len as usize);
+
+            ffi::g_checksum_get_digest(
+                mut_override(self.to_glib_none().0),
+                vec.to_glib_none().0,
+                &mut digest_len,
+            );
+
+            vec.set_len(digest_len);
+            vec
+        }
+    }
+
+    pub fn get_string(self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::g_checksum_get_string(
+                mut_override(self.to_glib_none().0),
+            ))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {Checksum, ChecksumType};
+
+    const CS_TYPE: ChecksumType = ChecksumType::Md5;
+    const CS_VALUE: &str = "fc3ff98e8c6a0d3087d515c0473f8677";
+
+    #[test]
+    fn update() {
+        let mut cs = Checksum::new(CS_TYPE);
+        cs.update("hello world!".as_bytes());
+        assert_eq!(cs.get_string().unwrap(), CS_VALUE);
+    }
+
+    #[test]
+    fn update_multi_call() {
+        let mut cs = Checksum::new(CS_TYPE);
+        cs.update("hello ".as_bytes());
+        cs.update("world!".as_bytes());
+        assert_eq!(cs.get_string().unwrap(), CS_VALUE);
+    }
+
+    #[test]
+    fn get_digest() {
+        let mut cs = Checksum::new(CS_TYPE);
+        cs.update("hello world!".as_bytes());
+        let vec = cs.get_digest();
+        assert_eq!(vec.len(), 16);
+        assert_eq!(vec[0], 0xfc);
+        assert_eq!(vec[1], 0x3f);
+        assert_eq!(vec[2], 0xf9);
+        assert_eq!(vec[3], 0x8e);
+        assert_eq!(vec[4], 0x8c);
+        assert_eq!(vec[5], 0x6a);
+        assert_eq!(vec[6], 0x0d);
+        assert_eq!(vec[7], 0x30);
+        assert_eq!(vec[8], 0x87);
+        assert_eq!(vec[9], 0xd5);
+        assert_eq!(vec[10], 0x15);
+        assert_eq!(vec[11], 0xc0);
+        assert_eq!(vec[12], 0x47);
+        assert_eq!(vec[13], 0x3f);
+        assert_eq!(vec[14], 0x86);
+        assert_eq!(vec[15], 0x77);
```

Comment:
For simplicity, you could define an `[u8]` array with all those bytes and then directly compare the two.